stdlib/LibGit2/src/gitcredential.jl

```diff
@@ -90,35 +79,45 @@ function Base.parse(::Type{GitCredential}, url::AbstractString)
 end
 
 function Base.copy!(a::GitCredential, b::GitCredential)
-    # Note: deepcopy calls avoid issues with securezero!
-    a.protocol = deepcopy(b.protocol)
-    a.host = deepcopy(b.host)
-    a.path = deepcopy(b.path)
-    a.username = deepcopy(b.username)
-    a.password = deepcopy(b.password)
+    a.protocol = b.protocol
+    a.host = b.host
+    a.path = b.path
+    a.username = b.username
+    a.password = b.password
     return a
 end
 
 function Base.write(io::IO, cred::GitCredential)
-    cred.protocol !== nothing && println(io, "protocol=", cred.protocol)
-    cred.host !== nothing && println(io, "host=", cred.host)
-    cred.path !== nothing && cred.use_http_path && println(io, "path=", cred.path)
-    cred.username !== nothing && println(io, "username=", cred.username)
-    cred.password !== nothing && println(io, "password=", cred.password)
+    cred.protocol !== nothing && write(io, "protocol=", cred.protocol, '\n')
+    cred.host !== nothing && write(io, "host=", cred.host, '\n')
+    cred.path !== nothing && cred.use_http_path && write(io, "path=", cred.path, '\n')
+    cred.username !== nothing && write(io, "username=", cred.username, '\n')
+    cred.password !== nothing && write(io, "password=", cred.password, '\n')
     nothing
 end
 
 function Base.read!(io::IO, cred::GitCredential)
     # https://git-scm.com/docs/git-credential#IOFMT
     while !eof(io)
-        key, value = split(readline(io), '=')
-
+        key = readuntil(io, '=')
+        if key == "password"
```

Comment:
We may also want to do this if `key == "url"` since that URL could contain a password

Reply:
We still parse that URL with a regex which stashes things into Strings.  I decided to just consider URLs insecure, but perhaps we should re-write that, too.

Reply:
That approach makes sense. I think we should add a comment which calls the potential security issue and leave it at that. The "url" key doesn't seem commonly used so it isn't a big concern for me.

Reply:
Any ideas as to where this would most make sense to document?  Another alternative would just be to completely _not support_ passwords in URLs.  We'd just throw an error if `m[:password]` exists.  I kinda like that.

Reply:
I think it makes sense to open an issue about this and reference the issue below `if key == "url"`. I believe we can address the issue later by rewriting the URL parsing and reading in the password with a `SecretBuffer`. We can try to tackle this now but I don't want to add extra work for you. 

Raising an error will still result in the password information being stored in a string which is no better than what we currently have.